packages/node-cli/src/commands/actionresult/DeployActionResult.js

```diff
@@ -0,0 +1,85 @@
+/*
+** Copyright (c) 2019 Oracle and/or its affiliates.  All rights reserved.
+** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+*/
+'use strict';
+const assert = require('assert');
+const ActionResult = require('./ActionResult');
+
+class DeployActionResult extends ActionResult {
+
+	constructor(build) {
+		super(build)
+		this._isServerValidation = build.isServerValidation ? true : false;
+		this._appliedContentProtection = build.appliedContentProtection ? true : false
+	}
+
+	validateBuild(build) {
+		assert(build);
+		assert(build.status, "status is required when creating an ActionResult object.");
+		if (build.status === ActionResult.SUCCESS) {
+			assert(build.data, "data is required when ActionResult is a success.");
+		}
+		if (build.status === ActionResult.ERROR) {
+			assert(build.errorMessages, "errorMessages is required when ActionResult is an error.");
+		}
```

Comment:
Ok good catch, I would delete this method here and it would use the one of Action Result

I would overrite completely the method in the rest of commands, because some of them don't use "data" (Proxy, SetupAccount), and if do so... they include other properties to check (CreateProject)

MOst of command anyways use the normal actionresult and have data, so it's ok to leave there the assert(build.data)

Reply:
Well, thinking twice, I'll try at least to use the super.validateBuild in CreateProjectActionResult